EIPS/eip-4944.md

```diff
@@ -50,22 +56,29 @@ In the constructor, automint the first token and set the variable to true:
     }
 ```
 
-Add additional functions to interact with the NFT properties (for instance, ERC20):
+Add additional functions to interact with the NFT properties (for instance, ERC-20):
 
 ``` solidity
     modifier onlyOwner() {
         require(balanceOf(msg.sender) > 0, "Caller is not the owner of the NFT");
         _;
     }
 
-    function transferTokens(IERC20 token, address recipient, uint256 amount) public onlyOwner {
+    function transferTokens(IERC20 token, address recipient, uint256 amount) public virtual onlyOwner {
         token.transfer(recipient, amount);
     }
+	
+    function balanceTokens(IERC20 token) public view virtual returns (uint256) {
+        return token.balanceOf(address(this));
+    }
 ```
 
+Here an example implementation: https://github.com/victormunoz/ERC-4944/blob/main/WalletNFT.sol
```

Comment:
If you'd like to include a reference implementation, you'll need to put it in the `../assets/eip-4944` directory and link to it there.